File: utils/python/json_parser.py
#!/usr/bin/env python3
# https://github.com/Zeronetsec/Chprompt

import os
import glob
import json

N = "\x1b[0m"
R = "\x1b[1;31m"
GG = "\x1b[0;32m"
CC = "\x1b[0;36m"
DG = "\x1b[1;90m"
WW = "\x1b[0;37m"

target_metadata = os.getenv("root")

if not target_metadata:
    print(f"{R}[!] {N}Variable: {GG}root {N}not found!")
    exit(1)

metadata_path = os.path.join(target_metadata, "metadata", "*.json")

for file_path in glob.glob(metadata_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            data = json.load(f)
            
            command = data.get("Command", "")
            args = data.get("Args", "")
            desc = data.get("Description", "")
            
            if args:
                fullcmd = f"{GG}{command} {CC}{args}{N}"
            else:
                fullcmd = f"{GG}{command}{N}"
                
            print(f"    {DG}* {fullcmd}")
            print(f"    {DG}└── {WW}{desc}{N}")
            
    except (json.JSONDecodeError, KeyError, PermissionError) as e:
        continue

# Copyright (c) 2026 Zeronetsec

File: utils/perl/json_parser.pl
#!/usr/bin/env perl
# https://github.com/Zeronetsec/Chprompt

use strict;
use warnings;
use File::Spec;
use JSON::PP;

my $target_metadata = $ENV{'root'};
if (!$target_metadata) {
    print "\x1b[1;31m[!] \x1b[0mVariable: \x1b[0;32mroot \x1b[0mnot found!\n";
    exit(1);
}

my $metadata_path = File::Spec->catfile(
    $target_metadata, "metadata", "*.json"
);

for my $file_path (glob($metadata_path)) {
    eval {
        open(
            my $fh, '<:encoding(UTF-8)', $file_path
        ) or die $!;

        local $/;
        my $json_text = <$fh>;
        close($fh);

        my $data = decode_json($json_text);

        my $command = $data->{Command} // "";
        my $args = $data->{Args} // "";
        my $desc = $data->{Description} // "";

        my $fullcmd;
        if ($args ne "") {
            $fullcmd = "\x1b[0;32m${command} \x1b[0;36m${args}\x1b[0m";
        } else {
            $fullcmd = "\x1b[0;32m${command}\x1b[0m";
        }

        print "    \x1b[1;90m* ${fullcmd}\n";
        print "    \x1b[1;90m└── \x1b[0;37m${desc}\x1b[0m\n";
    };

    if ($@) {
        next;
    }
}

# Copyright (c) 2026 Zeronetsec